File: FastExcelExportingDemo_src/FastExcelExportingDemoVb/ExportingCellByCellMethod.vb
Imports Microsoft.Office.Interop.Excel

Module ExportingCellByCellMethod

    Public Sub ExportToExcel(ByVal dataSet As DataSet, ByVal outputPath As String)
        ' Create the Excel Application object
        Dim excelApp As New ApplicationClass()

        ' Create a new Excel Workbook
        Dim excelWorkbook As Workbook = excelApp.Workbooks.Add(Type.Missing)

        Dim sheetIndex As Integer = 0
        Dim col, row As Integer
        Dim excelSheet As Worksheet

        ' Copy each DataTable as a new Sheet
        For Each dt As System.Data.DataTable In dataSet.Tables

            sheetIndex += 1

            ' Create a new Sheet
            excelSheet = CType( _
                excelWorkbook.Sheets.Add(excelWorkbook.Sheets(sheetIndex), _
                Type.Missing, 1, XlSheetType.xlWorksheet), Worksheet)

            excelSheet.Name = dt.TableName

            ' Copy the column names (cell-by-cell)
            For col = 0 To dt.Columns.Count - 1
                excelSheet.Cells(1, col + 1) = dt.Columns(col).ColumnName
            Next

            CType(excelSheet.Rows(1, Type.Missing), Range).Font.Bold = True

            ' Copy the values (cell-by-cell)
            For col = 0 To dt.Columns.Count - 1
                For row = 0 To dt.Rows.Count - 1
                    excelSheet.Cells(row + 2, col + 1) = dt.Rows(row).ItemArray(col)
                Next
            Next

            excelSheet = Nothing
        Next

        ' Save and Close the Workbook
        excelWorkbook.SaveAs(outputPath, XlFileFormat.xlWorkbookNormal, Type.Missing, _
         Type.Missing, Type.Missing, Type.Missing, XlSaveAsAccessMode.xlExclusive, _
         Type.Missing, Type.Missing, Type.Missing, Type.Missing, Type.Missing)

        excelWorkbook.Close(True, Type.Missing, Type.Missing)

        excelWorkbook = Nothing

        ' Release the Application object
        excelApp.Quit()
        excelApp = Nothing

        ' Collect the unreferenced objects
        GC.Collect()
        GC.WaitForPendingFinalizers()

    End Sub

End Module

File: RFT-Replaces/Backup17/ExportingCellByCellMethod.cs
using Microsoft.Office.Interop.Excel;
using System;
using System.Collections.Generic;
using System.Data;
using System.Text;

namespace FastExcelExportingDemoCs
{
	static class ExportingCellByCellMethod
	{

		public static void ExportToExcel(DataSet dataSet, string outputPath)
		{
			// Create the Excel Application object
			ApplicationClass excelApp = new ApplicationClass();

			// Create a new Excel Workbook
			Workbook excelWorkbook = excelApp.Workbooks.Add(Type.Missing);

			int sheetIndex = 0;

			// Copy each DataTable as a new Sheet
			foreach (System.Data.DataTable dt in dataSet.Tables)
			{

				// Create a new Sheet
				Worksheet excelSheet = (Worksheet) excelWorkbook.Sheets.Add(
					excelWorkbook.Sheets.get_Item(++sheetIndex),
					Type.Missing, 1, XlSheetType.xlWorksheet);

				excelSheet.Name = dt.TableName;

				// Copy the column names (cell-by-cell)
				for (int col = 0; col < dt.Columns.Count; col++) {
					excelSheet.Cells[1, col + 1] = dt.Columns[col].ColumnName;
				}

				((Range) excelSheet.Rows[1, Type.Missing]).Font.Bold = true;


				// Copy the values (cell-by-cell)
				for (int col = 0; col < dt.Columns.Count; col++)
				{
					for (int row = 0; row < dt.Rows.Count; row++)
					{
						excelSheet.Cells[row + 2, col + 1] = dt.Rows[row].ItemArray[col];
					}
				}

			}

			// Save and Close the Workbook
			excelWorkbook.SaveAs(outputPath, XlFileFormat.xlWorkbookNormal, Type.Missing,
				Type.Missing, Type.Missing, Type.Missing, XlSaveAsAccessMode.xlExclusive,
				Type.Missing, Type.Missing, Type.Missing, Type.Missing, Type.Missing);
			excelWorkbook.Close(true, Type.Missing, Type.Missing);
			excelWorkbook = null;

			// Release the Application object
			excelApp.Quit();
			excelApp = null;

			// Collect the unreferenced objects
			GC.Collect();
			GC.WaitForPendingFinalizers();

		}

	}

}
